### ISSUE TITLE
CreateInvoicesAsync(..) is not setting Reference field for ACCPAY types

### ISSUE THREAD

**Issue by timbochu** [XeroAPI/Xero-NetStandard#249]
- Xero.NetStandard.OAuth2 v3.7.2

When creating a Bill (ie ACCPAY type invoice) via CreateInvoicesAsync(..), the "Reference" field is not being honoured/retained.

var invoices = new Invoices();
invoices._Invoices = new List<Xero.NetStandard.OAuth2.Model.Accounting.Invoice>();

//Add a Receivables Invoice (Reference works fine!)
invoices._Invoices.Add(new Invoice()
{
  Type = Invoice.TypeEnum.ACCREC,
  Status = Invoice.StatusEnum.DRAFT,
  Date = new DateTime(2020,11,8),
  DueDate = new DateTime(2020,11,15),
  Contact = {any contact},
  Reference = "My Receivables Invoice Reference Here"
});

//Add a Bill (Reference does NOT work)
invoices._Invoices.Add(new Invoice()
{
  Type = Invoice.TypeEnum.ACCPAY,
  Status = Invoice.StatusEnum.DRAFT,
  Date = new DateTime(2020,11,8),
  DueDate = new DateTime(2020,11,15),
  Contact = {any contact},
  Reference = "My Bill Reference Here"
});

var task = Task.Run(async () => await api.CreateInvoicesAsync(myAccessToken, myTenantId, invoices));
task.Wait();

**Expected behavior**
The Bill (ACCPAY) item should show "My Bill Reference Here" in the reference field in Xero.

**Additional context**
Basically Xero isn't keeping the Reference string I'm sending through the CreateInvoicesAsync(..) method, but only when creating as Type=ACCPAY

**Comment by jenksguo** [comment]
Hi @timbochu, I understand this must be confusing for you. 

In Xero API, the **_Reference_** can only be used for account receivable (ACCREC) invoices. 

Our invoice documentation is [here](https://developer.xero.com/documentation/api/invoices). 
![image](https://user-images.githubusercontent.com/41350731/98926329-4700fa80-252b-11eb-8a38-675d5af5a6e1.png)

In account payable (ACCPAY) invoices, you can only use **_InvoiceNumber_** which it is displayed in UI as "Reference". 

![image](https://user-images.githubusercontent.com/41350731/98926597-a19a5680-252b-11eb-9491-341641eb4d2b.png)

I hope this clarifies the issue for you. :-D

**Comment by timbochu** [comment]
Awesome - thanks very much for the response. I've made the changes to code and will come back to you if there are any further issues. 
Have a good one!
